osu.Game/Beatmaps/Drawables/StarRatingDisplay.cs

```diff
@@ -0,0 +1,100 @@
+        private readonly Box background;
+        private readonly SpriteIcon starIcon;
+        private readonly OsuSpriteText starsText;
+
+        private readonly BindableWithCurrent<StarDifficulty> current = new BindableWithCurrent<StarDifficulty>();
+
+        public Bindable<StarDifficulty> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        [Resolved(canBeNull: true)]
+        private OverlayColourProvider colourProvider { get; set; }
+
+        /// <summary>
+        /// Creates a new <see cref="StarRatingDisplay"/> using an already computed <see cref="StarDifficulty"/>.
+        /// </summary>
+        /// <param name="starDifficulty">The already computed <see cref="StarDifficulty"/> to display.</param>
+        public StarRatingDisplay(StarDifficulty starDifficulty)
+        {
+            Current.Value = starDifficulty;
+
+            Size = new Vector2(52f, 20f);
+
+            InternalChild = new CircularContainer
+            {
+                Masking = true,
+                RelativeSizeAxes = Axes.Both,
+                Children = new Drawable[]
+                {
+                    background = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                    },
+                    starIcon = new SpriteIcon
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Margin = new MarginPadding { Right = 30f },
+                        Icon = FontAwesome.Solid.Star,
+                        Size = new Vector2(8f),
+                    },
+                    starsText = new OsuSpriteText
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Margin = new MarginPadding { Left = 10f, Bottom = 2f },
+                        // todo: this should be size: 12f, but to match up with the design, it needs to be 14.4f
+                        // see https://github.com/ppy/osu-framework/issues/3271.
+                        Font = OsuFont.Torus.With(size: 14.4f, weight: FontWeight.Bold),
+                        Shadow = false,
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Current.BindValueChanged(c =>
+            {
+                starsText.Text = c.NewValue.Stars.ToString("0.00");
+
+                background.Colour = colours.ForStarDifficulty(c.NewValue.Stars);
+
+                starIcon.Colour = c.NewValue.Stars >= 6.5 ? colours.Orange1 : colourProvider?.Background5 ?? Color4Extensions.FromHex("303d47");
+                starsText.Colour = c.NewValue.Stars >= 6.5 ? colours.Orange1 : colourProvider?.Background5 ?? Color4.Black.Opacity(0.75f);
```

Comment:
I was assuming the one in the beatmap info is the newest, and the one in the longue/songselect/otherwise is a fallback when using in non overlay contexts.

Though now that B5 is used for editor screens as well which is not an overlay, it sounds like B5 should be primarily used everywhere instead.